source/EventBus.Core.pas

```diff
@@ -304,30 +305,36 @@ procedure TEventBus.RegisterSubscriber<T>(ASubscriber: TObject; ARaiseExcIfEmpty
   try
     LSubscriberClass := ASubscriber.ClassType;
     LSubscriberMethods := TSubscribersFinder.FindSubscriberMethods<T>(LSubscriberClass, ARaiseExcIfEmpty);
-    for LSubscriberMethod in LSubscriberMethods do Subscribe<T>(ASubscriber, LSubscriberMethod);
+
+    for LSubscriberMethod in LSubscriberMethods do begin
+      // If the method do not have its Context specified, we assign the instance-specific context to it.
+      // Note - the instance-specific context itself may be empty.
+      if LSubscriberMethod.Context = '' then LSubscriberMethod.SetNewContext(AInstanceContext);

```

Comment:
we have to think here who is the master... If have set something using the Attribute I guess I want to override it with the InstanceContext.  So it becomes: 
```
if not AInstanceContext = '' then LSubscriberMethod.SetNewContext(AInstanceContext);
```
What do you think? My idea is that the runtime context should win over the attribute one